src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1815,7 +1817,8 @@ fn set_members_of_composite_type(cx: &CodegenCx<'ll, 'tcx>,
                     member_description.offset.bits(),
                     match member_description.discriminant {
                         None => None,
-                        Some(value) => Some(cx.const_u64(value)),
+                        Some(value) =>
+                            Some(cx.const_uint_big(cx.type_ix(align), value)),
```

Comment:
@eddyb This `align` might be bad, do you have any ideas?

Reply:
You just want `i128`, always, so `const_uint_big(cx.type_i128(), value)`.